tools/archetype/binding/src/main/resources/archetype-resources/src/main/java/internal/__bindingIdCamelCase__Handler.java

```diff
@@ -68,6 +73,11 @@ public void initialize() {
 
         // TODO: Initialize the thing. If done set status to ONLINE to indicate proper working.
         // Long running initialization should be done asynchronously in background.
+
+        // logger.debug("Start initializing!");
```

Comment:
This isn't clear to me - or rather I suspect that you have a slightly wrong understanding of the expectation.
Actually, the initialize method is supposed to return quickly and to set "some" state at the end - be it ONLINE or UNKNOWN, but it does not have to set it to UNKNOWN at the top of the method and to ONLINE at its end.

Reply:
@bjoernbrings Did you see this comment?